src/main/java/traincraft/blocks/assemblytables/AssemblyTableRecipeFactory.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Traincraft
+ * Copyright (c) 2011-2020.
+ * <p>
+ * This file ("AssemblyTableRecipeFactory.java") is part of the Traincraft mod for Minecraft.
+ * It is created by all people that are listed with @author below.
+ * It is distributed under LGPL-v3.0.
+ * You can find the source code at https://github.com/Traincraft/Traincraft
+ */
+
+package traincraft.blocks.assemblytables;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.JsonUtils;
+import net.minecraft.util.NonNullList;
+import net.minecraftforge.common.crafting.IRecipeFactory;
+import net.minecraftforge.common.crafting.JsonContext;
+import net.minecraftforge.oredict.OreDictionary;
+
+import javax.annotation.Nullable;
+
+import static net.minecraft.item.crafting.ShapedRecipes.deserializeItem;
+
+/**
+ * Json parser for all 3 tiers of assembly table.
+ *
+ * @author PseudonymPatel
+ * @since 2020-8-7
+ */
+public class AssemblyTableRecipeFactory implements IRecipeFactory {
+    
+    @Override
+    public IRecipe parse(JsonContext context, JsonObject json) {
+        
+        if (!json.has("tier")) {
+            throw new JsonSyntaxException("No tier element found in assembly table recipe");
+        }
+        int tier = json.get("tier").getAsInt();
+        
+        if (!json.has("result")) {
+            throw new JsonSyntaxException("No result element found in assembly table recipe");
+        }
+        ItemStack resultItemStack = deserializeItem(JsonUtils.getJsonObject(json, "result"), true);
+        
+        //set a blank AssemblyTableRecipe to add to.
+        AssemblyTableRecipe recipeInProgress = new AssemblyTableRecipe(tier, resultItemStack);
+        
+        if (json.has("planks")) {
+            recipeInProgress.setCraftingIngredient(0, deserializeIngredient(json.get("planks")));
+        } else {
+            throw new JsonSyntaxException("Could not find planks in assembly table recipe.");
+        }
+    
+        //we can iterate through an array of all the elements' names to set each of the 10 slots
+        String[] components = {"planks", "chimney", "cab", "dye", "component", "boiler", "firebox", "wheels", "frame", "coupler"};
```

Comment:
This doesn't change anywhere, so can you make it to a private static final variable?

Reply:
can't make private or static bc in method, so just made it final

Reply:
but you can move it outside of the method into the class as a private static final field